manifests/config.pp

```diff
@@ -120,5 +120,24 @@
         changes => template('foreman_proxy/sudo_augeas.erb'),
       }
     }
+  } else {
+    # The puppet-agent (puppet 4 AIO package) doesn't create a puppet user and group
+    # but the foreman proxy still needs to be able to read the agent's private key
+    if $foreman_proxy::manage_puppet_group {
+      if !defined(Group[$foreman_proxy::puppet_group]) {
+        group { $foreman_proxy::puppet_group:
+          ensure => 'present',
+        }
+      }
+      file { [
+        $foreman_proxy::ssldir,
+        "${foreman_proxy::ssldir}/private_keys",
+        $foreman_proxy::ssl_ca,
+        $foreman_proxy::ssl_key,
+        $foreman_proxy::ssl_cert ]:
+        group  => $foreman_proxy::puppet_group,
+        notify => Class['foreman_proxy::service'],
```

Comment:
yep.  Agreed, I'll get rid of this commit.  After my first commit, the service wasn't starting as it still couldn't read the keys.  The ordering wasn't the problem.  In the end, it was https://github.com/theforeman/puppet-foreman_proxy/pull/261/commits/cd725392e46e2dffa80c034309e7a464791706d5 that was needed.